test/hotspot/gtest/metaspace/test_is_metaspace_obj.cpp

```diff
@@ -49,7 +49,7 @@ class MetaspaceIsMetaspaceObjTest {
   }
 
   void do_test(Metaspace::MetadataType mdType) {
-    _lock = new Mutex(Monitor::native, "gtest-IsMetaspaceObjTest-lock", false, Monitor::_safepoint_check_never);
+    _lock = new Mutex(Monitor::leaf, "gtest-IsMetaspaceObjTest-lock", false, Monitor::_safepoint_check_never);
```

Comment:
Again I'm not at all clear on the basis for changing form the low ranked native to a much higher ranked leaf. ??

Reply:
Yes, it seemed easiest at the time. Replacing it with a ranked mutex is fine.

Reply:
I'm not at all sure why we would care about a gtest Mutex in terms of the ranking system?

Reply:
We could make it a PlatformMutex if we don't care about ranking.  Having the "native" rank available to bypass the ranking system is very confusing (why on earth is it > max_nonleaf, for instance?) and tempts misuse.  The ranking system has been long standing technical debt that everybody on the project has complained about at one time or another, and this is a small cleanup to remove one part of that.

Reply:
I guess "leaf" is as close to "not significant in terms of ranking" as we need to get in this case. I really don't like the idea of not using Mutexes just because we run into a ranking issue - PlatformMutex is meant as a building block not something for direct use. I guess gtest authors that need mutexes need to be careful about what they do while holding those mutexes.

Reply:
Yes, just like all other uses of Mutexes, you have to be aware of what locks can be taken while this lock is held.